components/eamxx/src/physics/dp/impl/dp_advance_iop_forcing_impl.hpp

```diff
@@ -13,10 +13,141 @@ namespace dp {
+  const Int nver_pack = ekat::npack<Spack>(nver);
+
+  // Set interface pressures
+  Kokkos::parallel_for(
+    Kokkos::TeamVectorRange(team, nver_pack), [&] (Int k) {
+      pint(k) = hyai(k)*ps0 + hybi(k)*ps;
+      pmid(k) = hyam(k)*ps0 + hybm(k)*ps;
+  });
+
+  // Set midpoint pressures and layer thicknesses
+  const auto pdel_s = scalarize(pdel);
+  const auto pint_s = scalarize(pint);
+  Kokkos::parallel_for(
+    Kokkos::TeamVectorRange(team, nver), [&] (Int k) {
+      pdel_s(k) = pint_s(k+1) - pint_s(k);
+  });
+}
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>::advance_iop_forcing(
+  // Input arguments
+  const Int& plev,
+  const Int& pcnst,
+  const bool& have_u,
+  const bool& have_v,
+  const bool& dp_crm,
+  const bool& use_3dfrc,
+  const Scalar& scm_dt,
+  const Scalar& ps_in,
+  const uview_1d<const Spack>& u_in,
+  const uview_1d<const Spack>& v_in,
+  const uview_1d<const Spack>& t_in,
+  const uview_2d<const Spack>& q_in,
+  const uview_1d<const Spack>& t_phys_frc,
+  const uview_1d<const Spack>& divt3d,
+  const uview_2d<const Spack>& divq3d,
+  const uview_1d<const Spack>& divt,
+  const uview_2d<const Spack>& divq,
+  const uview_1d<const Spack>& wfld,
+  const uview_1d<const Spack>& uobs,
+  const uview_1d<const Spack>& vobs,
+  const uview_1d<const Spack>& hyai,
+  const uview_1d<const Spack>& hyam,
+  const uview_1d<const Spack>& hybi,
+  const uview_1d<const Spack>& hybm,
+  // Kokkos stuff
+  const MemberType& team,
+  const Workspace& workspace,
+  // Output arguments
+  const uview_1d<Spack>& u_update,
+  const uview_1d<Spack>& v_update,
+  const uview_1d<Spack>& t_update,
+  const uview_2d<Spack>& q_update)
+{
+  // Local variables
+  uview_1d<Spack>
+    pmidm1, // pressure at model levels
+    pintm1, // pressure at model interfaces (dim=plev+1)
+    pdelm1; // pdel(k)   = pint  (k+1)-pint  (k)
+  workspace.template take_many_contiguous_unsafe<3>(
+    {"pmidm1", "pintm1", "pdelm1"},
+    {&pmidm1, &pintm1, &pdelm1});
+
+  // Get vertical level profiles
+  plevs0(plev, ps_in, hyai, hyam, hybi, hybm, team, pintm1, pmidm1, pdelm1);
+
+  constexpr Scalar rair = C::Rair;
+  constexpr Scalar cpair = C::Cpair;
+
+  ////////////////////////////////////////////////////////////
+  //  Advance T and Q due to large scale forcing
+
+  uview_1d<const Spack> t_lsf; // storage for temperature large scale forcing
+  uview_2d<const Spack> q_lsf; // storage for moisture large scale forcing
+
+  if (use_3dfrc) {
+    t_lsf = divt3d;
+    q_lsf = divq3d;
+  }
+  else {
+    t_lsf = divt;
+    q_lsf = divq;
+  }
+
+  const Int plev_pack = ekat::npack<Spack>(plev);
+  Kokkos::parallel_for(
+    Kokkos::TeamVectorRange(team, plev_pack), [&] (Int k) {
+      // Initialize thermal expansion term to zero.  This term is only
+      //  considered if using the preq-x dycore and if three dimensional
+      //  forcing is not provided by IOP forcing file.
+      Spack t_expan = 0;
+#ifndef MODEL_THETA_L
```

Comment:
I can't remember the result of the discussion, but don't we only use THETAL?

Reply:
That is correct.  In v0 this block of code was in place to provide support for the preqx dycore, which v1 does not have.  Thus, this (and other THETA_L if defs) can be removed.